mergify_engine/rules/__init__.py

```diff
@@ -256,19 +267,23 @@ async def create(
                     attr = condition.get_attribute_name()
                     if attr not in self.BASE_ATTRIBUTES:
                         condition.update("number>0")
+                    if attr in self.SPECIAL_ATTRIBUTES and (
+                        evaluated_rule not in self.special_attributes_rules
+                    ):
+                        self.special_attributes_rules.append(evaluated_rule)
```

Comment:
We should use the same logic as BASE_ATTRIBUTES for SPECIAL_ATTRIBUTES, to not just check if the special attribute is present, but if the binary tree can be true if the tree only has these attributes.

Imagine this tree: 
```
{"or": [
    "label: automerge",
    "base: foobar"
 ]}
 ```
 This rules should not be categorized in special_attributes_rules as we have a `or` with a label.